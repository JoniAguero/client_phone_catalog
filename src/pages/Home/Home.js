import React, { useEffect, useState } from "react"
import { getPhones } from "../../redux/actions/phonesActions"
import { useDispatch, useSelector } from "react-redux"
import './Home.css';
import Loading from "../../components/Loading";
import PlaceholderCard from "../../components/Loading/PlaceholderCard";
import CardPhone from "../../components/CardPhone";
import { Card } from "semantic-ui-react";
import NoDataFound from "../../components/NoDataFound";

export const Home = () => {

  const [loading, setLoading] = useState(true)
  const dispatch = useDispatch()
  const { phones } = useSelector( state => state )
  
  useEffect(() => {
    dispatch(getPhones())
    setTimeout(() => {
      setLoading(false)
    }, 1000);
  }, [dispatch])

  if ( loading ) {
    return (
      <div className="container-home">
        <Loading repetitions={2}>
          <PlaceholderCard />
        </Loading>
      </div> 
    );
  }

  return (
    <div className="container-home">
      <Card.Group itemsPerRow={3} stackable doubling>
        { 
          phones.list.length === 0 ? <NoDataFound /> : 
          phones.list.reverse().map(element => 
            <CardPhone phone={element} key={element._id}/>
          )
        }
      </Card.Group>
    </div>
  )
}