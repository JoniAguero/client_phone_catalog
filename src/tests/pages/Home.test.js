import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'

import Home from '../../pages/Home'

jest.mock('../../redux/actions/phonesActions', () => ({
  getPhones: jest.fn(),
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
  phones: {
    list: [],
    phoneSelected: null
  },
  errors: {
    error: {}
  }
}

const store = mockStore(initialState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <Home />
  </Provider>
)

describe('Home Page: ', () => {
  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
