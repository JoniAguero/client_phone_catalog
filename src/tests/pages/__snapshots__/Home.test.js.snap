// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page:  should be displayed correctly 1`] = `
<div
  className="container-home"
>
  <div
    className="container-loading"
  >
    <div
      key="0"
    >
      <div
        className="ui doubling stackable three cards"
      >
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      key="1"
    >
      <div
        className="ui doubling stackable three cards"
      >
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui card"
          onClick={[Function]}
        >
          <div
            className="ui placeholder"
          >
            <div
              className="square image"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="ui placeholder"
            >
              <div
                className="header"
              >
                <div
                  className="line very short"
                />
                <div
                  className="line medium"
                />
              </div>
              <div
                className="paragraph"
              >
                <div
                  className="line short"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
